File: app/src/main/java/com/example/mynewapplication/javapackage/ContentListActivity.java
package com.example.mynewapplication.javapackage;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.Toast;

import com.example.mynewapplication.javapackage.adapter.ContentListAdapter;
import com.example.mynewapplication.javapackage.adapter.MainAdapter;
import com.example.mynewapplication.javapackage.model.MainModel;
import com.example.mynewapplication.javapackage.model.ResponseModel;
import com.example.mynewapplication.javapackage.model.Results;
import com.example.mynewapplication.javapackage.viewmodel.MainActivityVM;
import com.example.mynewapplication.javapackage.viewmodel.MainActivityVMF;
import java.util.ArrayList;
import java.util.List;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.JsonArrayRequest;
import com.example.mynewapplication.R;



public class ContentListActivity extends AppCompatActivity {

    private String URL_JSON = "http://www.recipepuppy.com/api/";
    private JsonArrayRequest arrayRequest;
    private RequestQueue requestQueue;
    public ImageButton ib_back;
    public MainActivityVM viewModel;
    public MainActivityVMF viewModelFactory;
    private List<Results> resultsData = new ArrayList<>();
    private LinearLayoutManager layoutManager;
    private MainAdapter mainAdapter;
    private ArrayList<MainModel> list = new ArrayList<>();
    private MainModel mainModel;
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_content_list);
        viewModelFactory = new MainActivityVMF(this);
        viewModel = ViewModelProviders.of(ContentListActivity.this, viewModelFactory).get(MainActivityVM.class);

        ib_back = findViewById(R.id.ib_back);
        recyclerView = findViewById(R.id.rv_contentlist);

        ib_back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });


        initLiveData();
        initRecyclerView();
        viewModel.fetchRecipe(this);
    }

    public void initLiveData(){
        viewModel.getResults().observe(this, new Observer<ResponseModel>() {
            @Override
            public void onChanged(ResponseModel responseModel) {
                if (responseModel != null){
                    if (!resultsData.isEmpty()){
                        resultsData.clear();
                    }
                    resultsData = responseModel.getResults();
                    ContentListAdapter adapter = new ContentListAdapter(ContentListActivity.this, resultsData);
                    recyclerView.setAdapter(adapter);
                    adapter.notifyDataSetChanged();
                }else {
                    Toast.makeText(ContentListActivity.this, "No Result", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    public void initRecyclerView(){
        layoutManager = new LinearLayoutManager(ContentListActivity.this, recyclerView.VERTICAL, false);
        recyclerView.setLayoutManager(layoutManager);
        ContentListAdapter adapter = new ContentListAdapter(ContentListActivity.this, resultsData);
        recyclerView.setAdapter(adapter);
        adapter.notifyDataSetChanged();
    }

//    public void initLiveData(){
//
//    }
}

File: app/src/main/java/com/example/mynewapplication/ContentListActivity.kt
package com.example.mynewapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageButton
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.mynewapplication.Adapter.ContentListAdapter
import com.example.mynewapplication.Model.ListModel

class ContentListActivity : AppCompatActivity() {

    private lateinit var imageButton: ImageButton
    private lateinit var recyclerView: RecyclerView
    private lateinit var adapter: ContentListAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_content_list)
        imageButton = findViewById(R.id.ib_back)
        imageButton.setOnClickListener {
            startActivity(Intent(this, MainActivity::class.java))
        }

        init()

        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = adapter
    }

    private fun init() {
        recyclerView = findViewById(R.id.rv_contentlist)

        val data = ArrayList<ListModel>()
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))
        data.add(ListModel(R.drawable.food, "Food 1", "Food Desc", "Detail 1", "Detail 2", "Detail 3", "See Detail"))

        adapter = ContentListAdapter(data, object : ContentListAdapter.OnItemClickListener{
            override fun onClick(data: ListModel) {
                val intent = Intent(this@ContentListActivity, ContentDetailActivity::class.java)
                startActivity(intent)
            }

        })
//        adapter = ContentListAdapter(data, object : ContentListAdapter.OnAdapterListener{
//            override fun onClick(data: ListModel) {
////                startActivity(Intent(this, ContentDetailActivity::class.java))
//            }
//        })

    }

//    private fun sendDataToDetail(){
//        adapter.setOnclickListener(object : ContentListAdapter.OnItemClickListener{
//            override fun onClick(data: ListModel) {
//                val intent = Intent(this@ContentListActivity, ContentDetailActivity::class.java)
//                startActivity(intent)
//            }
//
//        })
    }